### PR TITLE
Res 1350 improvements weight decay

### PR DESCRIPTION
**Pull request by lucasosouza**
Added SGDW to PyTorch modules. Adapted (copied) from PyTorch pending pull requests.

### CODE REVIEW

File: nupic/research/frameworks/pytorch/modules/sgdw.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
```
Comment (subutai) [inline nupic/research/frameworks/pytorch/modules/sgdw.py:2]: 2020

File: nupic/research/frameworks/pytorch/modules/sgdw.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        return "<required parameter>"
+
+
+required = _RequiredParameter()
+
+
+class SGDW(Optimizer):
+    r"""Implements SGDW algorithm.
+
+    from: https://github.com/pytorch/pytorch/blob/
```
Comment (subutai) [inline nupic/research/frameworks/pytorch/modules/sgdw.py:38]: Just curious - is this copied as-is from PyTorch, or did you make modifications? Assuming yes, could you clarify in the comments what changes were made?  Could these changes be done as a subclass of the pytorch SGDW class instead of copying the whole thing?

Comment (lucasosouza) [reply nupic/research/frameworks/pytorch/modules/sgdw.py:38]: It is not yet implemented in pytorch. It is there as pull request, to be merged, so we can't pull it in.

File: nupic/research/frameworks/pytorch/modules/sgdw.py (hunk 1 above)
Comment (lucasosouza) [reply nupic/research/frameworks/pytorch/modules/sgdw.py]: fixed
